api/base/exceptions.py

```diff
@@ -11,7 +11,7 @@ def get_resource_object_member(error_key, context):
     if field:
         return 'relationships' if isinstance(field, RelationshipField) else 'attributes'
     # If field cannot be found (where read/write operations have different serializers,
-    # assume error was in 'attributes' by default
+    # or fields serialized on __init__, assume error was in 'attributes' by default
```

Comment:
Certain errors where we show the error source may not always be correct.  Because subjects are added to the serializer dynamically, there are issues trying to figure out where the error is in testing.  (An earlier commit in this PR initializes the serializer here instead of using `_declared_fields`, however this often causes a TransactionManagement error in testing.  I tried several things like [passing in transaction=True](https://pytest-django.readthedocs.io/en/latest/helpers.html#pytest-mark-django-db-request-database-access), but haven't found a good solution.  Deferring b/c there are only a few places where the field we want is not on `_declared_fields`, and it's not perfect currently, either. I need to think about this more.

For example, if I'm PATCHing a node's subjects, and I don't send in a list, my request might look like this: 
```json
PATCH localhost:8000/v2/nodes/9m4wp/?version=2.15
{
 "data": {
 	"id": "9m4wp",
 	"type": "nodes",
 	"relationships": {
 		"subjects": {
 			"data": {
 				"id": "afsjfaskljdf;akldsf",
 				"type": "subjects"
 					
 			}
 			
 		}
 		
 	}
 	
 }
}
```


My error message looks like this:
```json
{
    "errors": [
        {
            "source": {
                "pointer": "/data/attributes/subjects"
            },
            "detail": "Expected a list of items but got type \"unicode\"."
        }
    ],
    "meta": {
        "version": "2.15"
    }
}
```